siva/metadata.go

```diff
@@ -0,0 +1,278 @@
+package siva
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	billy "gopkg.in/src-d/go-billy.v4"
+	"gopkg.in/src-d/go-billy.v4/util"
+)
+
+const (
+	// LibraryMetadataFile is the name of the file that holds library metadata.
+	LibraryMetadataFile = "library.yaml"
+)
+
+// LibraryMetadata holds information about the library.
+type LibraryMetadata struct {
+	CurrentVersion int `json:"version"`
+
+	dirty bool
+}
+
+// NewLibraryMetadata creates a new LibraryMetadata.
+func NewLibraryMetadata(Version int) *LibraryMetadata {
+	return &LibraryMetadata{
+		CurrentVersion: Version,
+	}
+}
+
+// Version returns the version stored in the library metadata file or -1
+// if it's not set.
+func (m *LibraryMetadata) Version() int {
+	if m == nil {
+		return -1
+	}
+
+	return m.CurrentVersion
+}
+
+// SetVersion changes the current version.
+func (m *LibraryMetadata) SetVersion(v int) {
+	if v != m.CurrentVersion {
+		m.dirty = true
+		m.CurrentVersion = v
+	}
+}
+
+// Save writes metadata to the library yaml file.
+func (m *LibraryMetadata) Save(fs billy.Filesystem) error {
+	data, err := m.yaml()
+	if err != nil {
+		return err
+	}
+
+	tmp := LibraryMetadataFile + ".tmp"
+	defer fs.Remove(tmp)
+
+	err = util.WriteFile(fs, tmp, data, 0666)
+	if err != nil {
+		return err
+	}
+
+	return fs.Rename(tmp, LibraryMetadataFile)
+}
+
+// Dirty returns true if the metadata was changed and it needs to be written.
+func (m *LibraryMetadata) Dirty() bool {
+	return m.dirty
+}
+
+func (m *LibraryMetadata) yaml() ([]byte, error) {
```

Comment:
I think there's no need to wrap `yaml.Marshal` in a method instead of use it directly.